src/script/interpreter.cpp

```diff
@@ -1106,8 +1110,64 @@ class CTransactionSignatureSerializer {
 
 } // anon namespace
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion)
 {
+    if (sigversion == 1) {
+        uint256 hashPrevouts;
+        uint256 hashSequence;
+        uint256 hashOutputs;
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY)) {
+            CHashWriter ss(SER_GETHASH, 0);
+            for (unsigned int n = 0; n < txTo.vin.size(); n++) {
+                ss << txTo.vin[n].prevout;
+            }
+            hashPrevouts = ss.GetHash(); // TODO: cache this value for all signatures in a transaction
+        }
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY) && (nHashType & 0x1f) != SIGHASH_SINGLE && (nHashType & 0x1f) != SIGHASH_NONE) {
```

Comment:
Besides trying to make this version of the sig hash algorithm act similar to the original one, is there a reason why SINGLE and NONE are involved in this condition? To me SINGLE and NONE are flags controlling which output data gets committed to and it seems awkward that they also control which input sequences get committed. Just wondering if it makes sense to stop considering SINGLE and NONE here and then merge this loop with the previous one?